### ISSUE TITLE
splain leaves bare S<>

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11491]
Migrated from [rt.perl.org#94488](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=94488) (status was 'resolved')

Searchable as RT94488$

**Comment by p5pRT** [comment]
### From @cpansprout

$ perl5\.14\.0 \-Mdiagnostics \-e 'warn "syntax error"'
syntax error at \-e line 1 \(\#1\)
&nbsp;    \(F\) Probably means you had a syntax error\.  Common reasons include&#8203;:
&nbsp;    
&nbsp;        A keyword is misspelled\.
&nbsp;        A semicolon is missing\.
&nbsp;        A comma is missing\.
&nbsp;        An opening or closing parenthesis is missing\.
&nbsp;        An opening or closing brace is missing\.
&nbsp;        A closing quote is missing\.
&nbsp;    
&nbsp;    Often there will be another error message associated with the syntax
&nbsp;    error giving more information\.  \(Sometimes it helps to turn on \-w\.\)
&nbsp;    The error message itself often tells you where it was in the line when
&nbsp;    it decided to give up\.  Sometimes the actual error is several tokens
&nbsp;    before this\, because Perl is good at understanding random input\.
&nbsp;    Occasionally the line number may be misleading\, and once in a blue moon
&nbsp;    the only way to figure out what's triggering the error is to call
&nbsp;    perl \-c repeatedly\, chopping away half the program each time to see
&nbsp;    if the error went away\.  Sort of the cybernetic version of S\<20
&nbsp;    questions>\.

\(See that S\<20 questions> at the end? It doesn’t strip it even if it’s on one line\.\)



---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=low


---
Site configuration information for perl 5\.14\.0&#8203;:

Configured by sprout at Wed May 11 13&#8203;:45&#8203;:58 PDT 2011\.

Summary of my perl5 \(revision 5 version 14 subversion 0\) configuration&#8203;:
&nbsp;  Snapshot of&#8203;: eb70bb4a400e88a66c7e10414a2d52b5da4cfd1f
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=10\.5\.0\, archname=darwin\-thread\-multi\-2level
&nbsp;    uname='darwin pint\.local 10\.5\.0 darwin kernel version 10\.5\.0&#8203;: fri nov 5 23&#8203;:20&#8203;:39 pdt 2010; root&#8203;:xnu\-1504\.9\.17~1release\_i386 i386 '
&nbsp;    config\_args='\-Dusedevel \-de \-Duseithreads \-Doptimize=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-g'\,
&nbsp;    cppflags='\-fno\-common \-DPERL\_DARWIN \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 \(Apple Inc\. build 5664\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lutil \-lc
&nbsp;    perllibs=\-ldl \-lm \-lutil \-lc
&nbsp;    libc=/usr/lib/libc\.dylib\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    RC3



---
@&#8203;INC for perl 5\.14\.0&#8203;:
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.14\.0
&nbsp;    /usr/local/lib/perl5/5\.14\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/lib/perl5/5\.14\.0
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.



---
Environment for perl 5\.14\.0&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/sprout
&nbsp;    LANG=en\_US\.UTF\-8
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/bin&#8203;:/bin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/usr/X11/bin&#8203;:/usr/local/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @wolfsage

Patch attached\, also available here&#8203;:
https://github.com/wolfsage/perl/commit/7b2beb8

\.\.\. test suite still needed\. Hope to do this "soon"\.

\-\- Matthew Horsfall \(alh\)

**Comment by p5pRT** [comment]
### From @wolfsage

<details><summary>0001-RT-94488-Strip-S-codes-and-enforce-non-breaking-spac.patch</summary>

```diff
From 7b2beb8786095eb84c36a25319381d3711d190d8 Mon Sep 17 00:00:00 2001
From: Matthew Horsfall (alh) <wolfsage@gmail.com>
Date: Sun, 31 Jul 2011 23:26:03 -0400
Subject: [PATCH] RT #94488 - Strip S<> codes and enforce non-breaking spaces

---
 lib/diagnostics.pm |   10 ++++++++++
 lib/diagnostics.t  |    8 +++++++-
 2 files changed, 17 insertions(+), 1 deletions(-)

diff --git a/lib/diagnostics.pm b/lib/diagnostics.pm
index 60c96ea..3607d35 100644
--- a/lib/diagnostics.pm
+++ b/lib/diagnostics.pm
@@ -344,6 +344,11 @@ my %msg;
 	           ? italic($sect) . ' in ' . italic($page)
 	           : italic($page)
 	     /ges;
+            s/S<(.*?)>/
+               my $text = $1;
+               $text =~ s{[\r\n]}{ }gs;
+               $text;
+             /ges;
 	} else {
 	    s/C<<< (.*?) >>>|C<< (.*?) >>|[BC]<(.*?)>/$+/gs;
 	    s/[IF]<(.*?)>/$1/gs;
@@ -355,6 +360,11 @@ my %msg;
 	           ? qq '"$sect" in $page'
 	           : $page
 	     /ges;
+            s/S<(.*?)>/
+               my $text = $1;
+               $text =~ s{[\r\n]}{ }gs;
+               $text;
+             /ges;
 	} 
 	unless (/^=/) {
 	    if (defined $header) { 
diff --git a/lib/diagnostics.t b/lib/diagnostics.t
index 31a5802..5f0bcb8 100644
--- a/lib/diagnostics.t
+++ b/lib/diagnostics.t
@@ -5,7 +5,7 @@ BEGIN {
     @INC = 'lib';
 }
 
-use Test::More tests => 9;
+use Test::More tests => 10;
 
 BEGIN {
     my $w;
@@ -61,3 +61,9 @@ seek STDERR, 0,0;
 $warning = '';
 warn "Bad arg length for us, is 4, should be 42";
 like $warning, qr/In C parlance/, '%u works';
+
+# Non-breaking spaces
+seek STDERR, 0,0;
+$warning = '';
+warn "syntax error";
+like $warning, qr/cybernetic version of 20 questions/, 'non-breaking spaces';
-- 
1.7.0.4


```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout


On Jul 31\, 2011\, at 8&#8203;:31 PM\, Matthew Horsfall via RT wrote&#8203;:

> Patch attached\, also available here&#8203;:
> https://github.com/wolfsage/perl/commit/7b2beb8
> 
> \.\.\. test suite still needed\. Hope to do this "soon"\.
> 
> \-\- Matthew Horsfall \(alh\)
> 
> Patch attached\, also available here&#8203;: https://github.com/wolfsage/perl/commit/7b2beb8
> 
> \.\.\. test suite still needed\. Hope to do this "soon"\.
> 
> \-\- Matthew Horsfall \(alh\)

I think you are actually doing more in this patch than you need to\. Currently diagnostics\.pm does not do any text wrapping on its own\, but uses whatever is in perldiag\.pod\. So\, the S\<\.\.\.> sequence should never be wrapped in perldiag\.pod \(that 20 questions example should be fixed in perldiag\)\. All diagnostics\.pm should have to do is strip out the ‘S\<’ and the ‘>’\.

I you want to make diagnostics\.pm make sure the contents of S\<> are on one line\, then we should have it re\-wrap entire paragraphs\. But I think that would be overkill\.


> From 7b2beb8786095eb84c36a25319381d3711d190d8 Mon Sep 17 00&#8203;:00&#8203;:00 2001
> From&#8203;: Matthew Horsfall \(alh\) \<wolfsage@&#8203;gmail\.com>
> Date&#8203;: Sun\, 31 Jul 2011 23&#8203;:26&#8203;:03 \-0400
> Subject&#8203;: \[PATCH\] RT \#94488 \- Strip S\<> codes and enforce non\-breaking spaces
> 
> \-\-\-
> lib/diagnostics\.pm |   10 \+\+\+\+\+\+\+\+\+\+
> lib/diagnostics\.t  |    8 \+\+\+\+\+\+\+\-
> 2 files changed\, 17 insertions\(\+\)\, 1 deletions\(\-\)
> 
> diff \-\-git a/lib/diagnostics\.pm b/lib/diagnostics\.pm
> index 60c96ea\.\.3607d35 100644
> \-\-\- a/lib/diagnostics\.pm
> \+\+\+ b/lib/diagnostics\.pm
> @&#8203;@&#8203; \-344\,6 \+344\,11 @&#8203;@&#8203; my %msg;
>                ? italic\($sect\) \. ' in ' \. italic\($page\)
>                : italic\($page\)
>          /ges;
> \+            s/S\<\(\.\*?\)>/
> \+               my $text = $1;
> \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> \+               $text;
> \+             /ges;
>     \} else \{
>         s/C\<\<\< \(\.\*?\) >>>|C\<\< \(\.\*?\) >>|\[BC\]\<\(\.\*?\)>/$\+/gs;
>         s/\[IF\]\<\(\.\*?\)>/$1/gs;
> @&#8203;@&#8203; \-355\,6 \+360\,11 @&#8203;@&#8203; my %msg;
>                ? qq '"$sect" in $page'
>                : $page
>          /ges;
> \+            s/S\<\(\.\*?\)>/
> \+               my $text = $1;
> \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> \+               $text;
> \+             /ges;
>     \} 
>     unless \(/^=/\) \{
>         if \(defined $header\) \{ 
> diff \-\-git a/lib/diagnostics\.t b/lib/diagnostics\.t
> index 31a5802\.\.5f0bcb8 100644
> \-\-\- a/lib/diagnostics\.t
> \+\+\+ b/lib/diagnostics\.t
> @&#8203;@&#8203; \-5\,7 \+5\,7 @&#8203;@&#8203; BEGIN \{
>     @&#8203;INC = 'lib';
> \}
> 
> \-use Test&#8203;::More tests => 9;
> \+use Test&#8203;::More tests => 10;
> 
> BEGIN \{
>     my $w;
> @&#8203;@&#8203; \-61\,3 \+61\,9 @&#8203;@&#8203; seek STDERR\, 0\,0;
> $warning = '';
> warn "Bad arg length for us\, is 4\, should be 42";
> like $warning\, qr/In C parlance/\, '%u works';
> \+
> \+\# Non\-breaking spaces
> \+seek STDERR\, 0\,0;
> \+$warning = '';
> \+warn "syntax error";
> \+like $warning\, qr/cybernetic version of 20 questions/\, 'non\-breaking spaces';
> \-\- 
> 1\.7\.0\.4
>

**Comment by p5pRT** [comment]
### From @wolfsage

My mistake\, I misunderstood\.

Hopefully the attached patch is sufficient then :\)

Also found here&#8203;: https://github.com/wolfsage/perl/commit/113d9d11314bb9a

\-\- Matthew Horsfall \(alh\)

On Sun\, Aug 7\, 2011 at 3&#8203;:48 PM\, Father Chrysostomos \<sprout@&#8203;cpan\.org> wrote&#8203;:

>
> On Jul 31\, 2011\, at 8&#8203;:31 PM\, Matthew Horsfall via RT wrote&#8203;:
>
> > Patch attached\, also available here&#8203;:
> > https://github.com/wolfsage/perl/commit/7b2beb8
> >
> > \.\.\. test suite still needed\. Hope to do this "soon"\.
> >
> > \-\- Matthew Horsfall \(alh\)
> >
> > Patch attached\, also available here&#8203;:
> https://github.com/wolfsage/perl/commit/7b2beb8
> >
> > \.\.\. test suite still needed\. Hope to do this "soon"\.
> >
> > \-\- Matthew Horsfall \(alh\)
>
> I think you are actually doing more in this patch than you need to\.
> Currently diagnostics\.pm does not do any text wrapping on its own\, but
> uses whatever is in perldiag\.pod\. So\, the S\<\.\.\.> sequence should never be
> wrapped in perldiag\.pod \(that 20 questions example should be fixed in
> perldiag\)\. All diagnostics\.pm should have to do is strip out the ‘S\<’ and
> the ‘>’\.
>
> I you want to make diagnostics\.pm make sure the contents of S\<> are on one
> line\, then we should have it re\-wrap entire paragraphs\. But I think that
> would be overkill\.
>
>
> > From 7b2beb8786095eb84c36a25319381d3711d190d8 Mon Sep 17 00&#8203;:00&#8203;:00 2001
> > From&#8203;: Matthew Horsfall \(alh\) \<wolfsage@&#8203;gmail\.com>
> > Date&#8203;: Sun\, 31 Jul 2011 23&#8203;:26&#8203;:03 \-0400
> > Subject&#8203;: \[PATCH\] RT \#94488 \- Strip S\<> codes and enforce non\-breaking
> spaces
> >
> > \-\-\-
> > lib/diagnostics\.pm |   10 \+\+\+\+\+\+\+\+\+\+
> > lib/diagnostics\.t  |    8 \+\+\+\+\+\+\+\-
> > 2 files changed\, 17 insertions\(\+\)\, 1 deletions\(\-\)
> >
> > diff \-\-git a/lib/diagnostics\.pm b/lib/diagnostics\.pm
> > index 60c96ea\.\.3607d35 100644
> > \-\-\- a/lib/diagnostics\.pm
> > \+\+\+ b/lib/diagnostics\.pm
> > @&#8203;@&#8203; \-344\,6 \+344\,11 @&#8203;@&#8203; my %msg;
> >                  ? italic\($sect\) \. ' in ' \. italic\($page\)
> >                  : italic\($page\)
> >            /ges;
> > \+            s/S\<\(\.\*?\)>/
> > \+               my $text = $1;
> > \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> > \+               $text;
> > \+             /ges;
> >       \} else \{
> >           s/C\<\<\< \(\.\*?\) >>>|C\<\< \(\.\*?\) >>|\[BC\]\<\(\.\*?\)>/$\+/gs;
> >           s/\[IF\]\<\(\.\*?\)>/$1/gs;
> > @&#8203;@&#8203; \-355\,6 \+360\,11 @&#8203;@&#8203; my %msg;
> >                  ? qq '"$sect" in $page'
> >                  : $page
> >            /ges;
> > \+            s/S\<\(\.\*?\)>/
> > \+               my $text = $1;
> > \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> > \+               $text;
> > \+             /ges;
> >       \}
> >       unless \(/^=/\) \{
> >           if \(defined $header\) \{
> > diff \-\-git a/lib/diagnostics\.t b/lib/diagnostics\.t
> > index 31a5802\.\.5f0bcb8 100644
> > \-\-\- a/lib/diagnostics\.t
> > \+\+\+ b/lib/diagnostics\.t
> > @&#8203;@&#8203; \-5\,7 \+5\,7 @&#8203;@&#8203; BEGIN \{
> >     @&#8203;INC = 'lib';
> > \}
> >
> > \-use Test&#8203;::More tests => 9;
> > \+use Test&#8203;::More tests => 10;
> >
> > BEGIN \{
> >     my $w;
> > @&#8203;@&#8203; \-61\,3 \+61\,9 @&#8203;@&#8203; seek STDERR\, 0\,0;
> > $warning = '';
> > warn "Bad arg length for us\, is 4\, should be 42";
> > like $warning\, qr/In C parlance/\, '%u works';
> > \+
> > \+\# Non\-breaking spaces
> > \+seek STDERR\, 0\,0;
> > \+$warning = '';
> > \+warn "syntax error";
> > \+like $warning\, qr/cybernetic version of 20 questions/\, 'non\-breaking
> spaces';
> > \-\-
> > 1\.7\.0\.4
> >
>
>

**Comment by p5pRT** [comment]
### From @wolfsage

<details><summary>0001-Strip-S-formatting-codes-from-diagnostics-output-and.patch</summary>

```diff
From 113d9d11314bb9a1b92a66f63fd0a0e209596e71 Mon Sep 17 00:00:00 2001
From: Matthew Horsfall (alh) <wolfsage@gmail.com>
Date: Mon, 22 Aug 2011 21:26:26 -0400
Subject: [PATCH] Strip S<> formatting codes from diagnostics output and don't break S<20 questions> in perldiag.pod
 Strip S<> formatting codes from diagnostics output

Update spacing

Don't break S<20 questions> across lines, update test to be accurate
---
 lib/diagnostics.pm |    6 ++++++
 lib/diagnostics.t  |    7 ++++++-
 pod/perldiag.pod   |    3 +--
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/lib/diagnostics.pm b/lib/diagnostics.pm
index 7079d25..c462ada 100644
--- a/lib/diagnostics.pm
+++ b/lib/diagnostics.pm
@@ -344,6 +344,9 @@ my %msg;
 	           ? italic($sect) . ' in ' . italic($page)
 	           : italic($page)
 	     /ges;
+	     s/S<(.*?)>/
+               $1
+             /ges;
 	} else {
 	    s/C<<< (.*?) >>>|C<< (.*?) >>|[BC]<(.*?)>/$+/gs;
 	    s/[IF]<(.*?)>/$1/gs;
@@ -355,6 +358,9 @@ my %msg;
 	           ? qq '"$sect" in $page'
 	           : $page
 	     /ges;
+	    s/S<(.*?)>/
+               $1
+             /ges;
 	} 
 	unless (/^=/) {
 	    if (defined $header) { 
diff --git a/lib/diagnostics.t b/lib/diagnostics.t
index 31a5802..2ca92b0 100644
--- a/lib/diagnostics.t
+++ b/lib/diagnostics.t
@@ -55,9 +55,14 @@ $warning = '';
 warn "Execution of -e aborted due to compilation errors.\n";
 like $warning, qr/The final summary message/, 'Periods at end of line';
 
-
 # Test for %d/%u
 seek STDERR, 0,0;
 $warning = '';
 warn "Bad arg length for us, is 4, should be 42";
 like $warning, qr/In C parlance/, '%u works';
+
+# Strip S<>
+seek STDERR, 0,0;
+$warning = '';
+warn "syntax error";
+like $warning, qr/cybernetic version of 20 questions/s, 'strip S<>';
diff --git a/pod/perldiag.pod b/pod/perldiag.pod
index 97f16a4..88a63ec 100644
--- a/pod/perldiag.pod
+++ b/pod/perldiag.pod
@@ -4451,8 +4451,7 @@ before this, because Perl is good at understanding random input.
 Occasionally the line number may be misleading, and once in a blue moon
 the only way to figure out what's triggering the error is to call
 C<perl -c> repeatedly, chopping away half the program each time to see
-if the error went away.  Sort of the cybernetic version of S<20
-questions>.
+if the error went away.  Sort of the cybernetic version of S<20 questions>.
 
 =item syntax error at line %d: `%s' unexpected
 
-- 
1.7.0.4


```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Mon Aug 22 18&#8203;:59&#8203;:18 2011\, alh wrote&#8203;:
> My mistake\, I misunderstood\.
> 
> Hopefully the attached patch is sufficient then :\)
> 
> Also found here&#8203;: https://github.com/wolfsage/perl/commit/113d9d11314bb9a

Thank you\. Applied as 524e918\.

> 
> \-\- Matthew Horsfall \(alh\)
> 
> On Sun\, Aug 7\, 2011 at 3&#8203;:48 PM\, Father Chrysostomos \<sprout@&#8203;cpan\.org>
wrote&#8203;:
> 
> >
> > On Jul 31\, 2011\, at 8&#8203;:31 PM\, Matthew Horsfall via RT wrote&#8203;:
> >
> > > Patch attached\, also available here&#8203;:
> > > https://github.com/wolfsage/perl/commit/7b2beb8
> > >
> > > \.\.\. test suite still needed\. Hope to do this "soon"\.
> > >
> > > \-\- Matthew Horsfall \(alh\)
> > >
> > > Patch attached\, also available here&#8203;:
> > https://github.com/wolfsage/perl/commit/7b2beb8
> > >
> > > \.\.\. test suite still needed\. Hope to do this "soon"\.
> > >
> > > \-\- Matthew Horsfall \(alh\)
> >
> > I think you are actually doing more in this patch than you need to\.
> > Currently diagnostics\.pm does not do any text wrapping on its own\, but
> > uses whatever is in perldiag\.pod\. So\, the S\<\.\.\.> sequence should
never be
> > wrapped in perldiag\.pod \(that 20 questions example should be fixed in
> > perldiag\)\. All diagnostics\.pm should have to do is strip out the
‘S\<’ and
> > the ‘>’\.
> >
> > I you want to make diagnostics\.pm make sure the contents of S\<> are
on one
> > line\, then we should have it re\-wrap entire paragraphs\. But I think that
> > would be overkill\.
> >
> >
> > > From 7b2beb8786095eb84c36a25319381d3711d190d8 Mon Sep 17 00&#8203;:00&#8203;:00 2001
> > > From&#8203;: Matthew Horsfall \(alh\) \<wolfsage@&#8203;gmail\.com>
> > > Date&#8203;: Sun\, 31 Jul 2011 23&#8203;:26&#8203;:03 \-0400
> > > Subject&#8203;: \[PATCH\] RT \#94488 \- Strip S\<> codes and enforce non\-breaking
> > spaces
> > >
> > > \-\-\-
> > > lib/diagnostics\.pm |   10 \+\+\+\+\+\+\+\+\+\+
> > > lib/diagnostics\.t  |    8 \+\+\+\+\+\+\+\-
> > > 2 files changed\, 17 insertions\(\+\)\, 1 deletions\(\-\)
> > >
> > > diff \-\-git a/lib/diagnostics\.pm b/lib/diagnostics\.pm
> > > index 60c96ea\.\.3607d35 100644
> > > \-\-\- a/lib/diagnostics\.pm
> > > \+\+\+ b/lib/diagnostics\.pm
> > > @&#8203;@&#8203; \-344\,6 \+344\,11 @&#8203;@&#8203; my %msg;
> > >                  ? italic\($sect\) \. ' in ' \. italic\($page\)
> > >                  : italic\($page\)
> > >            /ges;
> > > \+            s/S\<\(\.\*?\)>/
> > > \+               my $text = $1;
> > > \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> > > \+               $text;
> > > \+             /ges;
> > >       \} else \{
> > >           s/C\<\<\< \(\.\*?\) >>>|C\<\< \(\.\*?\) >>|\[BC\]\<\(\.\*?\)>/$\+/gs;
> > >           s/\[IF\]\<\(\.\*?\)>/$1/gs;
> > > @&#8203;@&#8203; \-355\,6 \+360\,11 @&#8203;@&#8203; my %msg;
> > >                  ? qq '"$sect" in $page'
> > >                  : $page
> > >            /ges;
> > > \+            s/S\<\(\.\*?\)>/
> > > \+               my $text = $1;
> > > \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> > > \+               $text;
> > > \+             /ges;
> > >       \}
> > >       unless \(/^=/\) \{
> > >           if \(defined $header\) \{
> > > diff \-\-git a/lib/diagnostics\.t b/lib/diagnostics\.t
> > > index 31a5802\.\.5f0bcb8 100644
> > > \-\-\- a/lib/diagnostics\.t
> > > \+\+\+ b/lib/diagnostics\.t
> > > @&#8203;@&#8203; \-5\,7 \+5\,7 @&#8203;@&#8203; BEGIN \{
> > >     @&#8203;INC = 'lib';
> > > \}
> > >
> > > \-use Test&#8203;::More tests => 9;
> > > \+use Test&#8203;::More tests => 10;
> > >
> > > BEGIN \{
> > >     my $w;
> > > @&#8203;@&#8203; \-61\,3 \+61\,9 @&#8203;@&#8203; seek STDERR\, 0\,0;
> > > $warning = '';
> > > warn "Bad arg length for us\, is 4\, should be 42";
> > > like $warning\, qr/In C parlance/\, '%u works';
> > > \+
> > > \+\# Non\-breaking spaces
> > > \+seek STDERR\, 0\,0;
> > > \+$warning = '';
> > > \+warn "syntax error";
> > > \+like $warning\, qr/cybernetic version of 20 questions/\, 'non\-breaking
> > spaces';
> > > \-\-
> > > 1\.7\.0\.4
> > >
> >
> >

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Mon Aug 22 18&#8203;:59&#8203;:18 2011\, alh wrote&#8203;:
> My mistake\, I misunderstood\.
> 
> Hopefully the attached patch is sufficient then :\)
> 
> Also found here&#8203;: https://github.com/wolfsage/perl/commit/113d9d11314bb9a

Thank you\. Applied as 524e918\.

> 
> \-\- Matthew Horsfall \(alh\)
> 
> On Sun\, Aug 7\, 2011 at 3&#8203;:48 PM\, Father Chrysostomos \<sprout@&#8203;cpan\.org>
wrote&#8203;:
> 
> >
> > On Jul 31\, 2011\, at 8&#8203;:31 PM\, Matthew Horsfall via RT wrote&#8203;:
> >
> > > Patch attached\, also available here&#8203;:
> > > https://github.com/wolfsage/perl/commit/7b2beb8
> > >
> > > \.\.\. test suite still needed\. Hope to do this "soon"\.
> > >
> > > \-\- Matthew Horsfall \(alh\)
> > >
> > > Patch attached\, also available here&#8203;:
> > https://github.com/wolfsage/perl/commit/7b2beb8
> > >
> > > \.\.\. test suite still needed\. Hope to do this "soon"\.
> > >
> > > \-\- Matthew Horsfall \(alh\)
> >
> > I think you are actually doing more in this patch than you need to\.
> > Currently diagnostics\.pm does not do any text wrapping on its own\, but
> > uses whatever is in perldiag\.pod\. So\, the S\<\.\.\.> sequence should
never be
> > wrapped in perldiag\.pod \(that 20 questions example should be fixed in
> > perldiag\)\. All diagnostics\.pm should have to do is strip out the
‘S\<’ and
> > the ‘>’\.
> >
> > I you want to make diagnostics\.pm make sure the contents of S\<> are
on one
> > line\, then we should have it re\-wrap entire paragraphs\. But I think that
> > would be overkill\.
> >
> >
> > > From 7b2beb8786095eb84c36a25319381d3711d190d8 Mon Sep 17 00&#8203;:00&#8203;:00 2001
> > > From&#8203;: Matthew Horsfall \(alh\) \<wolfsage@&#8203;gmail\.com>
> > > Date&#8203;: Sun\, 31 Jul 2011 23&#8203;:26&#8203;:03 \-0400
> > > Subject&#8203;: \[PATCH\] RT \#94488 \- Strip S\<> codes and enforce non\-breaking
> > spaces
> > >
> > > \-\-\-
> > > lib/diagnostics\.pm |   10 \+\+\+\+\+\+\+\+\+\+
> > > lib/diagnostics\.t  |    8 \+\+\+\+\+\+\+\-
> > > 2 files changed\, 17 insertions\(\+\)\, 1 deletions\(\-\)
> > >
> > > diff \-\-git a/lib/diagnostics\.pm b/lib/diagnostics\.pm
> > > index 60c96ea\.\.3607d35 100644
> > > \-\-\- a/lib/diagnostics\.pm
> > > \+\+\+ b/lib/diagnostics\.pm
> > > @&#8203;@&#8203; \-344\,6 \+344\,11 @&#8203;@&#8203; my %msg;
> > >                  ? italic\($sect\) \. ' in ' \. italic\($page\)
> > >                  : italic\($page\)
> > >            /ges;
> > > \+            s/S\<\(\.\*?\)>/
> > > \+               my $text = $1;
> > > \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> > > \+               $text;
> > > \+             /ges;
> > >       \} else \{
> > >           s/C\<\<\< \(\.\*?\) >>>|C\<\< \(\.\*?\) >>|\[BC\]\<\(\.\*?\)>/$\+/gs;
> > >           s/\[IF\]\<\(\.\*?\)>/$1/gs;
> > > @&#8203;@&#8203; \-355\,6 \+360\,11 @&#8203;@&#8203; my %msg;
> > >                  ? qq '"$sect" in $page'
> > >                  : $page
> > >            /ges;
> > > \+            s/S\<\(\.\*?\)>/
> > > \+               my $text = $1;
> > > \+               $text =~ s\{\[\\r\\n\]\}\{ \}gs;
> > > \+               $text;
> > > \+             /ges;
> > >       \}
> > >       unless \(/^=/\) \{
> > >           if \(defined $header\) \{
> > > diff \-\-git a/lib/diagnostics\.t b/lib/diagnostics\.t
> > > index 31a5802\.\.5f0bcb8 100644
> > > \-\-\- a/lib/diagnostics\.t
> > > \+\+\+ b/lib/diagnostics\.t
> > > @&#8203;@&#8203; \-5\,7 \+5\,7 @&#8203;@&#8203; BEGIN \{
> > >     @&#8203;INC = 'lib';
> > > \}
> > >
> > > \-use Test&#8203;::More tests => 9;
> > > \+use Test&#8203;::More tests => 10;
> > >
> > > BEGIN \{
> > >     my $w;
> > > @&#8203;@&#8203; \-61\,3 \+61\,9 @&#8203;@&#8203; seek STDERR\, 0\,0;
> > > $warning = '';
> > > warn "Bad arg length for us\, is 4\, should be 42";
> > > like $warning\, qr/In C parlance/\, '%u works';
> > > \+
> > > \+\# Non\-breaking spaces
> > > \+seek STDERR\, 0\,0;
> > > \+$warning = '';
> > > \+warn "syntax error";
> > > \+like $warning\, qr/cybernetic version of 20 questions/\, 'non\-breaking
> > spaces';
> > > \-\-
> > > 1\.7\.0\.4
> > >
> >
> >

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
